invenio_records_files/api.py

```diff
@@ -283,10 +266,7 @@ def files(self):
             record_id=self.id).first()
 
         if not records_buckets:
-            bucket = self._create_bucket()
-            if not bucket:
-                return None
-            RecordsBuckets.create(record=self.model, bucket=bucket)
+            return None
```

Comment:
Could an empty iterator here, save us from edge cases? This way, one could go:

```python
for f in record.files:
  pass
```
directly without having to check for `None`. If there are still cases where `if record.files` is legitimately used, then `__bool__` on the `FileIterator` could address it.

Reply:
I think there's a difference here. ``None`` means no bucket is linked with the record, where ``[]`` means no files is in the bucket.

Reply:
I would side with separating these 2 meanings: 

- `record.files` just being about files (using a bucket behind the scenes, but never creating one)
- `record.bucket_id` (or `bucket` or `buckets` or `bucket_ids` :wink: ) being about whether the record is tied to a bucket(s). 

It would liberate mind space and prevent unintentioned coupling. I can see how it's easier to just use it like so for now though. I could make a pr for this if it makes sense after assessing the impact.